### ISSUE TITLE
Forward search does not work if building with separate outdir and auxdir

### ISSUE THREAD

**Issue by haoming-li-ling** [latex-lsp/texlab#940]
As in the title. I tried to set outdir and auxdir for latexmk separately. The auxDirectory option of texlab follows the auxdir option fed to latexmk. But then forward search doesn't work.

**Comment by pfoerster** [comment]
@haoming-li-ling Have you tried setting `texlab.build.pdfDirectory` to the same value as the `outdir` passed to `latexmk`?

**Comment by haoming-li-ling** [comment]
Ah, didn't see that option. However, with the following, forward search still doesn't work:
```lua
            auxDirectory = "./output",
            pdfDirectory = ".",
            build = {
              args = {
                "-lualatex",
                "-interaction=nonstopmode",
                "-synctex=1",
                "-auxdir=./output",
                "-outdir=.",
                "%f",
              },
```

**Comment by pfoerster** [comment]
Can you try adding an empty `.texlabroot` file to the directory containing the main TeX file? Maybe, the project detection isn't working here (see https://github.com/latex-lsp/texlab/wiki/Project-Detection)

**Comment by haoming-li-ling** [comment]
LspInfo reports the correct root directory for texlab; is this info sufficient?

**Comment by joedavid0** [comment]
I have the same issue. I have set `auxdir` to equal `texlab.build.auxDirectory` and the same for `outdir`, `texlab.build.logDirectory` and `texlab.build.pdfDirectory`, according to the wiki. Here are the relevant settings:
```lua
auxDirectory = ".aux/",
logDirectory = ".out/",
pdfDirectory = ".out/",
build = {
	args = {
		"-pdf",
		"-auxdir=./.aux",
		"-outdir=./.out",
		"-interaction=nonstopmode",
		"-synctex=1",
		"%f",
	},
},
```
`TexlabForward` just errors with `Search error`. When all the directories are the same, everything works as expected.

**Comment by pfoerster** [comment]
@joedavid0 
I think there is something wrong with your settings. It should be

```lua
build = {
        auxDirectory = ".aux/",
        logDirectory = ".out/",
        pdfDirectory = ".out/",
	args = {
		"-pdf",
		"-auxdir=./.aux",
		"-outdir=./.out",
		"-interaction=nonstopmode",
		"-synctex=1",
		"%f",
	},
},

```
Can you try again with these settings, please?

**Comment by joedavid0** [comment]
> Can you try again with these settings, please?

You're right, I missed that `auxDirectory` was moved under `build`. However, the issue still persists.
Texlab seems to look for the PDF in the project root, i.e. when I manually copy the PDF to `./`, Texlab opens it there.

**Comment by pfoerster** [comment]
@joedavid0 This might be a problem with the root directory detection. Can you try adding an empty `.texlabroot` file to the project? Then, it should look roughly like this:

```
<project folder>
   - .texlabroot
   - document.tex
   - .aux
       - document.aux
   - .out
       - document.log
       - document.pdf
```

**Comment by joedavid0** [comment]
Still the same problem

**Comment by pfoerster** [comment]
@joedavid0 Can you try running #988 and post the additional log statements (they start with `DEBUG - [FwdSearch]`)? These should be helpful to find out why the server looks into the wrong directory.

**Comment by joedavid0** [comment]
Ok this is kind embarrassing, but the error was on my part. The PR didn't have the issue, so I checked when the changes to config were introduced. Then I noticed that the version of Texlab on Arch Linux is surprisingly very outdated and doesn't have the new options yet... so that was the issue.

Sorry for wasting your time
